src/backend/users/management/commands/migrate_housing_seeker_data.py

```diff
@@ -0,0 +1,488 @@
+        return json.loads(profile_content)
+
+    def create_random_string(self, length=10):
+        chars = string.ascii_uppercase + string.digits
+        return "".join(random.choice(chars) for _ in range(length))
+
+    def is_key_val_nonempty(self, profile, field):
+        if field not in profile:
+            return False
+        if isinstance(profile[field], (str, list)):
+            return len(profile[field]) > 0
+        if isinstance(profile[field], (int, float, bool)):
+            return True
+        return False
+
+    def get_first_last_names(self, profile):
+        # base case: `clientEmail` is the ultimate fallback
+        first_name = profile[ProfileKeys.EMAIL]
+        last_name = profile[ProfileKeys.EMAIL]
+
+        # if `headOfHousehold` exists and not empty, use this name
+        if self.is_key_val_nonempty(profile, ProfileKeys.NAME):
+            names = profile[ProfileKeys.NAME].strip().split(" ")
+            if len(names) == 1:
+                # use the only word as the names
+                first_name = profile[ProfileKeys.NAME]
+                last_name = profile[ProfileKeys.NAME]
+            else:
+                # use the first word as first name; rest as last name
+                first_name = names[0]
+                last_name = "".join(names[1:])
+
+        return first_name, last_name
+
+    # Priority levels are consistent throughout all profile data
+    # The range is from 1 to 4 with increasing importance
+    def decode_priority(self, priority):
+        if priority == 1 or priority == "1":
+            return UserProfile.Priorities.NOT_IMPORTANT
+        elif priority == 2 or priority == "2":
+            return UserProfile.Priorities.SOMEWHAT_IMPORTANT
+        elif priority == 3 or priority == "3":
+            return UserProfile.Priorities.IMPORTANT
+        else:
+            return UserProfile.Priorities.VERY_IMPORTANT
+
+    # Among user profiles with emails
+    # no profile has both fields True
+    def decode_travel_mode(self, use_commuter_rail, has_vehicle):
+        if has_vehicle is True and use_commuter_rail is False:
+            return UserProfile.TravelMode.CAR
+
+        if has_vehicle is False and use_commuter_rail is True:
+            return UserProfile.TravelMode.BUS_TRAIN_EXPRESS
+
+        return UserProfile.TravelMode.BUS_TRAIN
+
+    def save_user_record(self, bucket, key, profile):
+        user = None
+        has_email = self.is_key_val_nonempty(profile, ProfileKeys.EMAIL)
+
+        # only email field is really used to migrate user records
+        # headOfHousehold is used for names, but email is the final
+        # fallback if headOfHousehold is empty
+        if has_email:
+            first_name, last_name = self.get_first_last_names(profile)
+            last_modified_at = self.get_s3_object_metadate(bucket, key)[self.S3_OBJ_TIMESTAMP_KEY]
+            password = self.create_random_string()
```

Comment:
I'd add a comment here to note that users will not be using passwords to log in, and this is just to ensure that passwords are not blank and not all the same, but we don't expect to use it.

Reply:
Good point. Addressed here: https://github.com/azavea/echo-locator/pull/478/commits/a36a8f148ac0ae9312f701d016466514c1a7c864